modules/tensor_mechanics/src/kernels/InertialForce.C

```diff
@@ -88,23 +102,87 @@ InertialForce::computeQpResidual()
     return _test[_i][_qp] * _density[_qp] *
            (accel + vel * _eta[_qp] * (1 + _alpha) - _alpha * _eta[_qp] * (*_vel_old)[_qp]);
   }
+
+  // Lumped mass option
+  // Only lumping the masses here
+  // will multiply by corresponding residual multiplier after lumping the matrix
+  else if (_time_integrator->isLumped() && _time_integrator->isExplicit())
+    return _test[_i][_qp] * _density[_qp];
+
+  // Consistent mass option
+  // Same for explicit, implicit, and implicit with HHT
   else
     return _test[_i][_qp] * _density[_qp] *
-           ((*_u_dotdot)[_qp] + (*_u_dot)[_qp] * _eta[_qp] * (1.0 + _alpha) -
+           ((*_u_dotdot_residual)[_qp] + (*_u_dot_residual)[_qp] * _eta[_qp] * (1.0 + _alpha) -
             _alpha * _eta[_qp] * (*_u_dot_old)[_qp]);
 }
 
+void
+InertialForce::computeResidual()
+{
+  prepareVectorTag(_assembly, _var.number());
+
+  precalculateResidual();
+  for (_i = 0; _i < _test.size(); _i++)
+    for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+    {
+      // Here, computeQpResidual only contains the qp mass when lumped mass
+      // formulation is used. Therefore, the line below adds the qp masses,
+      // which is mass lumping.
+      _local_re(_i) += _JxW[_qp] * _coord[_qp] * computeQpResidual();
+    }
+
+  // Residual calculation for lumped-mass matrices for explicit integration
+  if (_time_integrator->isLumped() && _time_integrator->isExplicit())
+  {
+    std::vector<const Node *> node;
+    node.resize(_test.size());
+    for (unsigned int i = 0; i < node.size(); ++i)
+      node[i] = _current_elem->node_ptr(i);
+
+    // Fetch the solution for the nodes in the at time t
+    NonlinearSystemBase & nonlinear_sys = _fe_problem.getNonlinearSystemBase();
+    const NumericVector<Number> & u_dotdot_residual = _time_integrator->uDotDotResidual();
+    const NumericVector<Number> & u_dot_residual = _time_integrator->uDotResidual();
+    Real u_dot_residual_node, u_dotdot_residual_node;
+    for (unsigned int j = 0; j < node.size(); j++)
+    {
+      u_dot_residual_node =
```

Comment:
`_variable->dofIndices()` can be used here to grab the current indices.

However, accessing these nodal residual values really should be done through some interface if possible.

You shouldn't need to get the nodes and the dof numbers like you are. I believe `_variable->dofIndices()` should be available for the residual calls.